crates/core/tedge_write/src/bin.rs

```diff
@@ -30,9 +32,23 @@ pub struct Args {
     /// Group which will become the new owner of the file.
     #[arg(long)]
     group: Option<Box<str>>,
+
+    #[command(flatten)]
+    log_args: LogConfigArgs,
+
+    /// [env: TEDGE_CONFIG_DIR, default: /etc/tedge]
+    #[clap(
+        long = "config-dir",
+        default_value = tedge_config::get_config_dir().into_os_string(),
+        hide_env_values = true,
+        hide_default_value = true,
+    )]
+    config_dir: Utf8PathBuf,
```

Comment:
As @reubenmiller proposed in on of the earlier PRs, wouldn't it make sense to capture these two arguments into a common struct and use a flattened reference to it in all these concrete structs like `MapperOpt`, AgentOpt`?

Reply:
We could put `config_dir` into the common struct as well, but because it's a multicall binary we'd still have to include it in `Opt` struct for each component and can't really put it in a single place and make all the other `Opts` have references to it. They have to own it directly.

Reply:
I think I may have wrongly used the word "reference" here, which caused that confusion. What I meant was to have a flattened field, something like:

```
    #[command(flatten)]
    common_args: CommonArgs,
```

and not really a "reference" to a single instance of that struct, which probably would unnecessarily complicated the code.